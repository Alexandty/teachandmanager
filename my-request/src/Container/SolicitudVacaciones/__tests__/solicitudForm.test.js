import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { SolicitudForm } from '../index';
import ConnectedSolicitudForm from '../index';
import { Button } from 'react-bootstrap';
import configureStore from 'redux-mock-store';

const mockStore = configureStore()

configure({ adapter: new Adapter() });

describe('Probando vista SolicitudForm', () => {
    it('Validando toMatchSnapshot', () => {
        const props = {
            loadAvailableDays: jest.fn(),
            guardar: jest.fn(),
            consultar: jest.fn(),
            handleSubmit: jest.fn(),
            avalableDaysData: 0,
            availableDaysVacation: true,
            user: { user: 'a' }
        };
        const wrap = shallow(<SolicitudForm {...props} />);
        expect(toJson(wrap)).toMatchSnapshot();
    });

    it('Probar la accion consulta de vacaciones', () => {
        const spyGuardar = jest.fn();
        const props = {
            avalableDaysData: 0,
            availableDaysVacation: true,
            consultar: jest.fn(),
            loadAvailableDays: jest.fn(),
            guardar: spyGuardar, //validar el login
            handleSubmit: (guardar) => {
                //fake de la respuesta
                guardar({ username: 'raul', password: 'alzate' });
            },
            pristine: true,
            user: { user: 'a' }
        };
        const wrap = shallow(<SolicitudForm {...props} />);
        wrap.find(Button).simulate('click')

        expect(spyGuardar).toBeCalledWith({ username: 'raul', password: 'alzate', user: 'a' });
    });

    it('renders the connected component ConnectedSolicitudForm', () => {
        const initialState = {}
        const store = mockStore(initialState)
        const wrapper = shallow(<ConnectedSolicitudForm store={store} />);
    });
});