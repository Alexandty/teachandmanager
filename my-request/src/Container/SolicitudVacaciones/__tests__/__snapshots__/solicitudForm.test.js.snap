// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando vista SolicitudForm Validando toMatchSnapshot 1`] = `
<div>
  <h2>
    Solicitud
  </h2>
  <form>
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        className="show-grid"
        componentClass="div"
      >
        <div>
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
            xs={6}
          >
            Dias disponibles 
            <Label
              bsClass="label"
              bsStyle="primary"
            >
               
              0
               
            </Label>
            <FormGroup
              bsClass="form-group"
              controlId="formInlineDate"
            >
              <Label
                bsClass="label"
                bsStyle="default"
              >
                Fecha de Inicio
              </Label>
              <Field
                component={[Function]}
                id="idStartDate"
                name="startDate"
                type="Date"
              />
            </FormGroup>
            <FormGroup
              bsClass="form-group"
              controlId="formInlineDate"
            >
              <Label
                bsClass="label"
                bsStyle="default"
              >
                Fecha de Fin
              </Label>
              <Field
                component={[Function]}
                id="idEndtDate"
                name="endDate"
                type="Date"
              />
            </FormGroup>
            <Alert
              bsClass="alert"
              bsStyle="info"
              closeLabel="Close alert"
            />
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="success"
              disabled={true}
              type="submit"
            >
              Solicitar
            </Button>
          </Col>
        </div>
      </Row>
    </Grid>
  </form>
</div>
`;
